packages/space-opera/src/components/mobile_view/mobile_view.ts

```diff
@@ -50,102 +47,107 @@ export class MobileView extends ConnectedLitElement {
     } else if (this.config.environmentImage) {
-      partialState.config.environmentImage = this.config.environmentImage;
+      snippet.config.environmentImage = this.config.environmentImage;
     }
 
-    reduxStore.dispatch(dispatchSetHotspots(partialState.hotspots));
-    reduxStore.dispatch(dispatchSetCamera(partialState.camera));
-    reduxStore.dispatch(dispatchSetConfig(partialState.config));
-    reduxStore.dispatch(dispatchArConfig(partialState.arConfig));
-
-    await this.updateComplete;
+    this.hotspots = snippet.hotspots;
+    this.arConfig = snippet.arConfig;
+    this.config = snippet.config;
+    this.camera = snippet.camera;
 
     // Send a new POST out for each scene-viewer button press
-    if (partialState.arConfig.ar) {
+    if (snippet.arConfig.ar) {
       const arButton =
           this.modelViewer?.shadowRoot!.getElementById('default-ar-button')!;
       // @ts-ignore
       arButton.addEventListener('click', (event: MouseEvent) => {
-        this.postSceneViewerBlob(this.currentBlob!);
+        post(this.currentBlob!, this.modelViewerUrl);
       });
     }
   }
 
-  async waitForEnv(envIsHdr: boolean) {
-    const response = await fetch(this.envPipeUrl);
+  // TODO: Fix iOS not loading USDZ.
+  async waitForUSDZ(usdzId: number) {
+    const response =
+        await fetch(usdzToSession(this.pipeId, this.sessionId, usdzId));
     if (response.ok) {
       const blob = await response.blob();
-      // simulating createBlobUrlFromEnvironmentImage
-      const addOn = envIsHdr ? '#.hdr' : '';
-      const envUrl = URL.createObjectURL(blob) + addOn;
-      reduxStore.dispatch(dispatchEnvrionmentImage(envUrl));
+      const usdzUrl = URL.createObjectURL(blob);
+      this.arConfig.iosSrc = usdzUrl;
+    } else {
+      console.error('Error:', response);
     }
   }
 
-  async waitForUSDZ(modelIds: number) {
-    const response = await fetch(
-        `https://ppng.io/modelviewereditor-usdz-${this.pipeId}-${modelIds}`);
+  async waitForEnv(envIsHdr: boolean) {
+    const response = await fetch(envToSession(this.pipeId, this.sessionId));
     if (response.ok) {
+      // Simulating createBlobUrlFromEnvironmentImage
       const blob = await response.blob();
-      const usdzUrl = URL.createObjectURL(blob);
-      reduxStore.dispatch(dispatchIosSrc(usdzUrl));
-    } else {
-      console.error('Error:', response);
+      const addOn = envIsHdr ? '#.hdr' : '';
+      const envUrl = URL.createObjectURL(blob) + addOn;
+      this.envImageUrl = envUrl;
     }
   }
 
-  async waitForData(json: any) {
-    if (json.iosChanged) {
-      await this.waitForUSDZ(json.modelIds);
-      await this.updateComplete;
+  // We set modelViewerUrl instead of directly fetching it because scene-viewer
+  // requires the same url from the current model-viewer state, and we need to
+  // make a POST request to that URL when scene-viewer is triggered.
+  // TODO: Look into if that is the same issue with quick-look.
```

Comment:
This seems like the correct approach in general, even for the environment, right?

Reply:
Yes, just tested and it works for environment image as well.